docs/benchmarkdotnet.md

```diff
@@ -0,0 +1,332 @@
+
+The glob patterns are applied to full benchmark name: namespace.typeName.methodName. Examples:
+
+* `dotnet run -f netcoreapp3.0 --filter BenchmarksGame*` - will run all the benchmarks from BenchmarksGame namespace
+* `dotnet run -f netcoreapp3.0 --filter *.Richards.*` - will run all the benchmarks with type name Richards
+* `dotnet run -f netcoreapp3.0 --filter *.ToStream` - will run all the benchmarks with method name ToStream
+* `dotnet run -f netcoreapp3.0 --filter *` - **will run ALL benchmarks**
+
+You can provide many filters (logical disjunction):
+
+```cmd
+dotnet run -f netcoreapp3.0 --filter System.Collections*.Dictionary* *.Perf_Dictionary.*
+```
+
+To print a **joined summary** for all of the benchmarks (by default printed per type), use `--join`:
+
+```cmd
+dotnet run -f netcoreapp2.1 --filter BenchmarksGame* --join
+```
+
+Please remember that on **Unix** systems `*` is resolved to all files in current directory, so you need to escape it `'*'`. The filter is **case insensitive**.
+
+#### Listing the Benchmarks
+
+To print the list of all available benchmarks you need to pass `--list [tree/flat]` argument. It can also be combined with `--filter` option.
+
+Example: Show the tree of all the benchmarks from System.Threading namespace that can be run for .NET Core 2.0:
+
+```cmd
+dotnet run -f netcoreapp2.0 --list tree --filter System.Threading*
+```
+
+```log
+System
+ └─Threading
+    ├─Channels
+    │  └─Tests
+    │     ├─BoundedChannelPerfTests
+    │     │  ├─TryWriteThenTryRead
+    │     │  ├─WriteAsyncThenReadAsync
+    │     │  ├─ReadAsyncThenWriteAsync
+    │     │  └─PingPong
+    │     ├─SpscUnboundedChannelPerfTests
+    │     │  ├─TryWriteThenTryRead
+    │     │  ├─WriteAsyncThenReadAsync
+    │     │  ├─ReadAsyncThenWriteAsync
+    │     │  └─PingPong
+    │     └─UnboundedChannelPerfTests
+    │        ├─TryWriteThenTryRead
+    │        ├─WriteAsyncThenReadAsync
+    │        ├─ReadAsyncThenWriteAsync
+    │        └─PingPong
+    ├─Tasks
+    │  ├─ValueTaskPerfTest
+    │  │  ├─Await_FromResult
+  ..... // the list continues
+```
+
+## Reading the Results
+
+BenchmarkDotNet prints all important statistics to the console:
+
+```log
+Mean = 429.6484 ns, StdErr = 2.8826 ns (0.67%); N = 18, StdDev = 12.2296 ns
+Min = 419.6885 ns, Q1 = 422.2286 ns, Median = 425.6508 ns, Q3 = 428.5053 ns, Max = 459.5134 ns
+IQR = 6.2768 ns, LowerFence = 412.8134 ns, UpperFence = 437.9205 ns
+ConfidenceInterval = [418.2188 ns; 441.0781 ns] (CI 99.9%), Margin = 11.4297 ns (2.66% of Mean)
+Skewness = 1.46, Kurtosis = 3.57, MValue = 2
+```
+
+Due to the limited space in the console, only Mean, Standard Error and Standard Deviation are printed in the table:
+
+| Method |     Mean |    Error |   StdDev |
+|------- |---------:|---------:|---------:|
+|  Adams | 429.6 ns | 11.43 ns | 12.23 ns |
+
+### Reading the Histogram
+
+Each Iteration is represented by `@`:
+
+```
```

Comment:
`MD040/fenced-code-language: Fenced code blocks should have a language specifiedmarkdownlint(MD040)`

Reply:
You should use ```log